src/domain/app/header/userDropdown/UserDropdown.tsx

```diff
@@ -32,11 +32,15 @@ const UserDropdown: React.FunctionComponent = () => {
         {
           label: t('authentication.logout.text'),
           onClick: () => {
+            // Clear user form data
             dispatch(resetFormValues());
-            // Clear user form data from redux store
-            persistor.flush();
+            // Clear profile (fetched from API)
+            dispatch(clearProfile());
+            // Clear backend auth data
+            dispatch(fetchTokenError());
             // Flush data in redux store and localStorage
-            logoutTunnistamo(location.pathname);
+            persistor.flush();
+            logoutTunnistamo();
```

Comment:
Should user be redirected to some specific page after logout?

Reply:
Good question! If you are on a page that requires authentication, you will end up on the front page automatically, since we have an automatic redirect for the restricted paths. 

Do you think we should add an explicit redirect or just a comment here explaining?

Reply:
Ah, I guess it's fine like this if the router keeps track of that.